src/server/util/file_io_adaptor.h

```diff
@@ -0,0 +1,70 @@
+/** Copyright 2020-2022 Alibaba Group Holding Limited.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+The implementation of `get_rss`, `get_shared_rss` and `get_peek_rss`
+are referred from https://stackoverflow.com/a/14927379/5080177, which
+has the following license header originally:
+
+  * Author:  David Robert Nadeau
+  * Site:    http://NadeauSoftware.com/
+  * License: Creative Commons Attribution 3.0 Unported License
+  *          http://creativecommons.org/licenses/by/3.0/deed.en_US
```

Comment:
There's no `get_rss`, ... in this file so please double check the copyright header if you were doing some copy-paste jobs.